java-manta-client-unshaded/src/main/java/com/joyent/manta/config/BaseChainedConfigContext.java

```diff
@@ -121,6 +121,11 @@
      */
     private volatile Integer skipDirectoryDepth;
 
+    /**
+     * Whether or not we can attempt to resume a download automatically.
+     */
+    private volatile Integer downloadContinuations;
```

Comment:
Using volatile strikes me as a little odd for a non-primitive type. In particular an integer. If you were looking for a thread-safe nullable integer type, normally (for good or bad) - I tend to reach for `AtomicInteger` or a volatile primitive type. I'm not saying this is wrong, but I would double check the semantics and the need for a non-primitive type.

Reply:
I was following the existing pattern based on our decision in [this comment](https://github.com/joyent/java-manta/pull/357#issuecomment-340869194) when we made the ConfigContext reloadable. I opted for an `Integer` since the default was was originally `null` for "disable continuations" but since we have an explicit off value (`0`) now we can make it a primitive `int`.